addOns/pscanrulesBeta/src/test/java/org/zaproxy/zap/extension/pscanrulesBeta/InformationDisclosureSuspiciousCommentsUnitTest.java

```diff
@@ -0,0 +1,270 @@
+import java.nio.file.Paths;
+import java.util.Arrays;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Test;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+public class InformationDisclosureSuspiciousCommentsUnitTest
+        extends PassiveScannerTest<InformationDisclosureSuspiciousComments> {
+
+    @Override
+    protected InformationDisclosureSuspiciousComments createScanner() {
+        return new InformationDisclosureSuspiciousComments();
+    }
+
+    @Override
+    public void setUpZap() throws Exception {
+        super.setUpZap();
+
+        Path xmlDir =
+                Files.createDirectories(
+                        Paths.get(
+                                Constant.getZapHome(),
+                                InformationDisclosureSuspiciousComments.suspiciousCommentsListDir));
+        Path testFile =
+                xmlDir.resolve(InformationDisclosureSuspiciousComments.suspiciousCommentsListFile);
+        Files.write(testFile, Arrays.asList("# FixMeNot", "  FixMe  ", "TODO", "\t "));
+    }
+
+    protected HttpMessage createHttpMessageWithRespBody(String responseBody, String contentType)
+            throws HttpMalformedHeaderException, URIException {
+
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setURI(new URI("http://example.com", false));
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+        msg.setResponseBody(responseBody);
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n"
+                        + "Server: Apache-Coyote/1.1\r\n"
+                        + "Content-Type: "
+                        + contentType
+                        + "\r\n"
+                        + "Content-Length: "
+                        + responseBody.length()
+                        + "\r\n");
+        return msg;
+    }
+
+    @Test
+    public void shouldReadSuspiciousCommentsFile() {
+
+        // When
+        File suspiciousCommentsFile =
+                new File(
+                        Constant.getZapHome()
+                                + File.separator
+                                + InformationDisclosureSuspiciousComments.suspiciousCommentsListDir
+                                + File.separator
+                                + InformationDisclosureSuspiciousComments
+                                        .suspiciousCommentsListFile);
+
+        // Then
+        assertTrue(
+                "Couldn't find "
+                        + InformationDisclosureSuspiciousComments.suspiciousCommentsListFile
+                        + " file at: "
+                        + suspiciousCommentsFile.getAbsolutePath(),
+                suspiciousCommentsFile.exists());
```

Comment:
I don't think we need to assert that the file exists, which case(s) is this intended to catch? (It's the test that's creating the file so any problem would be "caught" before this assertion?)

Reply:
This test was copied from another similar test case that did the same thing. It's very useful to verify that the configuration file is properly being loaded when running the test case. For example, before we changed how it was loaded specifically for this test case, this test case was failing.  I recommend it be left in.

Reply:
> This test was copied from another similar test case that did the same thing.

Ok, will check why the other test is doing that, if it's like this case it seems odd.

> It's very useful to verify that the configuration file is properly being loaded when running the test cases.

It's not verifying that, it's merely asserting that the file exists (which obviously will since the test set up is creating it, otherwise this test and thus the assertion would not be executed).

> For example, before we changed how it was loaded specifically for this test case, this test case was failing. I recommend it be left in.

That does not show anything wrong with the scanner just that the tests were not properly set up.

Reply:
Like I said in an earlier comment, this is all new to us, and its not obvious that when running the tests, they load their config files from a different place than it does normally, so adding the test case was our first confirmation that when running the test, the config file was not loading. We then fixed the test case so it did, and this test case now passes. So, as I said before. We'd prefer to live it in, but if you really want it out, we can remove it. We were just coding by example, as I said before.

Reply:
Its understandable that this is all new to you, and there are many things that are difficult to learn just from looking at the code.
I can see why you added this test to help you understand how the file was loaded - I often do similar things, usually with println statements ;)
But that doesnt mean that the code you used to help you understand the code should be converted into a test.
I defer to @thc202 's recommendations as they are always very well thought out, unless I feel I have a compelling counter argument :)

Reply:
OK. We'll wait to hear from him on my last two comments and then we'll make hopefully the final fixes.

Reply:
I still don't think that we need this test, it's not testing the behaviour of the scanner and for the reasons @psiinon said.